yt/frontends/ramses/tests/test_outputs.py

```diff
@@ -13,589 +14,494 @@
+        for field, _ in extra_fields:
+            assert ("all", field) in ds.field_list
+        # Check the family (they should equal 100, for tracer particles)
+        dd = ds.all_data()
+        families = dd[("all", "particle_family")]
+        assert all(families == 100)
+
+    @requires_file(ramsesExtraFieldsSmall)
+    def test_extra_fields_2(self):
+        extra_fields = [f"particle_extra_field_{i + 1}" for i in range(2)]
+        ds = yt.load(os.path.join(ramsesExtraFieldsSmall, "info_00001.txt"))
+        # the dataset should contain the fields
+        # In the dataset, the fields are integers, so we cannot test
+        # that they are accessed correctly.
+        for field in extra_fields:
+            assert ("io", field) in ds.field_list
+
+    @pytest.mark.parametrize("ds", [ramses_rt], indirect=True)
+    def test_ramses_rt(self, ds):
+        ad = ds.all_data()
+        expected_fields = [
+            "Density",
+            "x-velocity",
+            "y-velocity",
+            "z-velocity",
+            "Pres_IR",
+            "Pressure",
+            "Metallicity",
+            "HII",
+            "HeII",
+            "HeIII",
+        ]
+        for field in expected_fields:
+            assert ("ramses", field) in ds.field_list
+            # test that field access works
+            ad["ramses", field]
+        # test that special derived fields for RT datasets work
+        special_fields = [("gas", "temp_IR")]
+        species = ["H_p1", "He_p1", "He_p2"]
+        for specie in species:
+            special_fields.extend(
+                [
+                    ("gas", specie + "_fraction"),
+                    ("gas", specie + "_density"),
+                    ("gas", specie + "_mass"),
+                ]
+            )
+        for field in special_fields:
+            assert field in ds.derived_field_list
+            ad[field]
+
+    @requires_file(ramses_sink)
+    @requires_file(output_00080)
+    def test_ramses_sink(self):
+        expected_fields = [
+            "BH_bondi_accretion",
+            "BH_eddington_accretion",
+            "BH_efficiency",
+            "BH_esave",
+            "BH_real_accretion",
+            "BH_spin",
+            "BH_spin_x",
+            "BH_spin_y",
+            "BH_spin_z",
+            "gas_spin_x",
+            "gas_spin_y",
+            "gas_spin_z",
+            "particle_birth_time",
+            "particle_identifier",
+            "particle_mass",
+            "particle_position_x",
+            "particle_position_y",
+            "particle_position_z",
+            "particle_prop_0_0",
+            "particle_prop_0_1",
+            "particle_prop_0_2",
+            "particle_prop_0_3",
+            "particle_prop_0_4",
+            "particle_prop_0_5",
+            "particle_prop_0_6",
+            "particle_velocity_x",
+            "particle_velocity_y",
+            "particle_velocity_z",
+        ]
+        # Check that sinks are autodetected
```

Comment:
I know this isn't you, but wouldn't it be possible to parametrize this test (with `@pytest.mark.parametrize("ds", [ramses_sink, output_00080], indirect=True)`) and remove the code duplication below?